database/database.go

```diff
@@ -79,15 +79,15 @@ func ApplyConnectionsConfig(db *sql.DB, connections *ConnectionsConfig, logger l
 		db.SetMaxIdleConns(connections.MaxIdle)
 	}
 
-	if connections.MaxLifetime > 0 {
-		logger.Logf("setting SQL max lifetime to %v", connections.MaxLifetime)
-		db.SetConnMaxLifetime(connections.MaxLifetime)
-	}
-
 	if connections.MaxIdleTime > 0 {
 		logger.Logf("setting SQL max idle time to %v", connections.MaxIdleTime)
 		db.SetConnMaxIdleTime(connections.MaxIdleTime)
 	}
 
+	if connections.MaxLifetime > 0 {
+		logger.Logf("setting SQL max lifetime to %v", connections.MaxLifetime)
+		db.SetConnMaxLifetime(connections.MaxLifetime)
```

Comment:
Can we include a comment about how the order is critical? I never would have guessed the ordering would have made a difference. 

https://github.com/golang/go/issues/45993#issuecomment-1427873850

Reply:
yah makes sense, will do